backend/zkppSalt.js

```diff
@@ -0,0 +1,229 @@
+import { blake32Hash } from '~/shared/functions.js'
+import { timingSafeEqual } from 'crypto'
+import nacl from 'tweetnacl'
+import sbp from '@sbp/sbp'
+import { boxKeyPair, encryptContractSalt, hashStringArray, hashRawStringArray, hash, parseRegisterSalt, randomNonce, computeCAndHc } from '~/shared/zkpp.js'
+
+// TODO HARDCODED VALUES
+const recordPepper = 'pepper'
+const recordMasterKey = 'masterKey'
+const challengeSecret = 'secret'
+const registrationSecret = 'secret'
+const maxAge = 30
+
+const getZkppSaltRecord = async (contract: string) => {
+  const recordId = blake32Hash(hashStringArray('RID', contract, recordPepper))
+  const record = await sbp('chelonia/db/get', recordId)
+
+  if (record) {
+    const encryptionKey = hashStringArray('REK', contract, recordMasterKey).slice(0, nacl.secretbox.keyLength)
+
+    const recordBuf = Buffer.from(record.replace(/_/g, '/').replace(/-/g, '+'), 'base64')
```

Comment:
Thanks, I think the best course of action here is to call a function actually.
Background: There's base64 and base64url, and the difference between them is that base64url is 'URL-safe' so it doesn't require escaping when encoded as a URL. Node.js supports `Buffer.from(x, 'base64url')` and `.toString('base64url')` but not so the Buffer shim that is added to the compiled code. Hence, to use base64url, as is more appropriate here, the replacements are needed. Since they are in a few places, I'll call a function doing them.